lib/jormungandr/test/bench/Latency.hs

```diff
@@ -101,30 +111,106 @@ import Test.Integration.Framework.DSL
 
 main :: forall t n. (t ~ Jormungandr, n ~ 'Testnet) => IO ()
 main = do
     tvar <- newTVarIO []
     logging <- setupLatencyLogging tvar
 
-    let meanAvg :: [NominalDiffTime] -> Double
-        meanAvg ts = sum (map realToFrac ts) * 1000 / fromIntegral (length ts)
-    let buildResult :: [NominalDiffTime] -> Builder
-        buildResult [] = "ERR"
-        buildResult ts = build $ fixedF 1 $ meanAvg ts
-    let fmtResult :: String -> [NominalDiffTime] -> IO ()
-        fmtResult title ts = fmtLn ("    "+|title|+" - "+|buildResult ts|+" ms")
+    fmtLn "Latencies for 2 fixture wallets scenario"
+    runScenario logging tvar (nFixtureWallet 2)
 
-    fmtLn "Latencies for two fixture wallets scenario"
+    fmtLn "Latencies for 10 fixture wallets scenario"
+    runScenario logging tvar (nFixtureWallet 10)
 
-    withUtf8Encoding $ benchWithServer logging $ \ctx -> do
-        (wal1, wal2) <- twoFixtureWallet ctx
+    fmtLn "Latencies for 100 fixture wallets scenario"
+    runScenario logging tvar (nFixtureWallet 100)
+
+    fmtLn "Latencies for 2 fixture wallets with 10 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 2 10)
+
+    fmtLn "Latencies for 2 fixture wallets with 20 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 2 20)
+
+    fmtLn "Latencies for 2 fixture wallets with 100 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 2 100)
+
+    fmtLn "Latencies for 10 fixture wallets with 10 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 10 10)
+
+    fmtLn "Latencies for 10 fixture wallets with 20 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 10 20)
+
+    fmtLn "Latencies for 10 fixture wallets with 100 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 10 100)
+  where
+    nFixtureWallet n ctx = do
+        wal1 : wal2 : _ <- replicateM n (fixtureWallet ctx)
+        pure (wal1, wal2)
+
+    nFixtureWalletWithTxs n m ctx = do
+        (wal1, wal2) <- nFixtureWallet n ctx
+
+        let amt = (1 :: Natural)
+        let batchSize = 10
+        let whole10Rounds = div m batchSize
+        let lastBit = mod m batchSize
+        let amtExp val = fromIntegral ((fromIntegral val) + faucetAmt) :: Natural
+        let expInflows =
+                if whole10Rounds > 0 then
+                    [x*batchSize | x<-[1..whole10Rounds]] ++ [lastBit]
+                else
+                    [lastBit]
+        mapM_ (repeatPostTx ctx wal1 amt batchSize . amtExp) expInflows
+
+        pure (wal1, wal2)
+
+    repeatPostTx ctx wDest amtToSend batchSize amtExp = do
+        wSrc <- fixtureWallet ctx
+        let pass = "cardano-wallet" :: Text
+
+        replicateM_ batchSize (postTx ctx (wSrc, postTxEp, pass) wDest amtToSend)
+
+        rWal1 <- request @ApiWallet ctx (getWalletEp wDest) Default Empty
+        verify rWal1
+            [ expectSuccess
+            , expectEventually ctx getWalletEp balanceAvailable amtExp
```

Comment:
I haven't tried myself, but does `expectEventually` inside the `mapM_` cause fixture setup to take longer than it should?
How does this function ensure that the correct number of transactions were accepted?

Reply:
`expectEventually` will wait until balance of the recepient wallet will be as specified in `amtExp`. And here I am waiting for recepient wallet to have certain number following from batchSize=1 and amt=1 every run. So for 20 txs I am waiting in the first run for the wallet to have `faucetAmt + 10` (and hence 10 txs in tx history) and in the second run `faucetAmt + 10` (and hence 10 txs in tx history).